app/jobs/regular/shared_conversation_adjust_upload_security.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ::Jobs
+  class SharedConversationAdjustUploadSecurity < ::Jobs::Base
+    def execute(args)
+      # The conversation context includes post cooked content so this
+      # must be updated when target uploads security changes.
+      update_conversation(args[:conversation_id]) if args[:conversation_id].present?
+
+      # If we deleted the conversation then we just need to update the target's
+      # uploads security, no need to update the conversation.
+      if args[:target_id].present? && args[:target_type].present?
+        update_target(args[:target_id], args[:target_type])
+      end
+    end
+
+    def update_conversation(conversation_id)
+      conversation = SharedAiConversation.find(conversation_id)
+      return if conversation.blank?
```

Comment:
I believe `SharedAiConversation` is an ActiveRecord object, so `#find` would already have raised an error here. Perhaps we change to `SharedAiConversation.find_by(id: conversation_id)`?

Reply:
Yes you're right, I think I missed this from an earlier refactor :+1: